src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp

```diff
@@ -129,17 +129,33 @@ ov::Output<ov::Node> FixInputNodeRank(ov::Output<ov::Node> input_node, ov::Rank:
 }  // namespace
 
 namespace sink_forward {
+AxisVector AlignTransposeOrder(const Output<Node>& output, const TransposeInputsInfo& transpose_input_info) {
+    auto num_of_val = static_cast<int64_t>(shape_size(transpose_input_info.transpose_const->get_shape()));
+    const auto rank = output.get_partial_shape().rank().get_length();
+
+    AxisVector new_transpose_order;
+    if (rank > num_of_val) {
+        const auto diff = rank - num_of_val;
+        new_transpose_order.resize(rank);
+        std::iota(new_transpose_order.begin(), new_transpose_order.end(), 0);
+        auto transpose_axis_order = transpose_input_info.transpose_const->get_axis_vector_val();
+        for (int64_t i = diff; i < rank; ++i) {
+            new_transpose_order[i] = transpose_axis_order[i - diff] + diff;
+        }
```

Comment:
I am worried. Could you explain what are the use cases for such transpose?

Reply:
disscused offline, it will be fixed for Concat op

Reply:
the current implementation doesn't support dynamic ranks for Concat inputs, in this way this method should never add this alignment. 
dynamic inputs to Concat will be processed separately in the next PR